AwarenessApisSampleKotlin/app/src/main/java/com/android/example/awarenessapissamplekotlin/MainActivity.kt

```diff
@@ -37,6 +55,83 @@ class MainActivity : AppCompatActivity() {
 
     fun onClickSnapshot(view: View) {
         Log.d(TAG, "onClickSnapshot()")
+
+        if (activityRecognitionPermissionApproved()) {
+            requestSnapshot()
+        } else {
+            requestPermissionLauncher.launch(Manifest.permission.ACTIVITY_RECOGNITION)
+        }
+    }
+
+    private fun requestSnapshot() {
+        Log.d(TAG, "requestSnapshot()")
+
+        val task: Task<DetectedActivityResponse> =
+            Awareness.getSnapshotClient(this).detectedActivity
+
+        task.addOnCompleteListener { taskResponse: Task<DetectedActivityResponse> ->
+
+            if (taskResponse.isSuccessful) {
+                val detectedActivityResponse: DetectedActivityResponse = taskResponse.result
+                val activityRecognitionResult: ActivityRecognitionResult =
```

Comment:
maybe simplify to `val activityRecognitionResult = detectedActivityResponse.activityRecognitionResult`